components/reusable-components/tags-modal/TagsModal.tsx

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+
+'use client';
+
+import React, { ChangeEvent, useState } from 'react';
+import styles from './TagsModal.module.scss';
+
+interface TagSearchProps {
+  existingTags: string[];
+  selectedTags: string[];
+  setSelectedTags: (items: string[]) => void;
+  onClose: () => void;
+}
+
+const TagsModal = ({ existingTags, selectedTags, setSelectedTags, onClose }: TagSearchProps) => {
+  const [searchTag, setSearchTag] = useState<string>('');
+  const [filteredTags, setFilteredTags] = useState<string[]>([]);
+
+  const handleSearchChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const query = e.target.value;
+    setSearchTag(query);
+
+    if (query) {
+      const filtered = existingTags.filter((tag) =>
+        tag.toLowerCase().includes(query.toLowerCase())
+      );
+      setFilteredTags(filtered);
+    } else {
+      setFilteredTags([]);
+    }
+  };
+
+  const addTag = (tag: string) => {
+    if (!selectedTags.includes(tag)) {
+      setSelectedTags([...selectedTags, tag]);
+    }
+    setSearchTag('');
+    setFilteredTags([]);
+  };
+
+  const removeTag = (tag: string) => {
+    setSelectedTags(selectedTags.filter((item) => item !== tag));
+  };
+
+  return (
+    <div className={styles.modalWrapper}>
+      <div className={styles.modalContent}>
+        <button type="button" className={styles.closeButton} onClick={onClose}>
+          <i aria-label="Close button" className="bi bi-x" />
+        </button>
+        {/* eslint-disable jsx-a11y/label-has-associated-control  */}
```

Comment:
lets try using this in the eslint configuration:
{
  "rules": {
    "jsx-a11y/label-has-associated-control": [
      "error",
      {
        "assert": "htmlFor",
        "depth": 3
      }
    ]
  }
}

Reply:
Its trying to enfore us to use it like:
<label><input/> </label>
using htmlFor with same value as the id should fix that, but since it is still shouting at us, lets update it as I mentioned above.